src/Entities/MessageEntity.php

```diff
@@ -18,23 +18,22 @@ class MessageEntity extends Entity
     protected $url;
 
     /**
-     * @todo check for value from this list: https://core.telegram.org/bots/api#messageentity
-     * @todo "empty($this->offset) && $this->offset != 0" - not an ideal solution?
+     * @todo check for type value from this list: https://core.telegram.org/bots/api#messageentity
      */
     public function __construct(array $data)
     {
         $this->type = isset($data['type']) ? $data['type'] : null;
-        if (empty($this->type)) {                               //
+        if (empty($this->type)) {
             throw new TelegramException('type is empty!');
         }
 
         $this->offset = isset($data['offset']) ? $data['offset'] : null;
-        if (empty($this->offset) && $this->offset != 0) {       //
+        if ($this->offset === '') {
```

Comment:
Ok, so it can be `0`. And clearly you've tested this, yes? 😊 

Sorry, I'm sometimes a bit paranoid about changes like this, as they sometimes might introduce the little edge cases that produce weird errors.